apiserver/facades/client/cloud/cloud.go

```diff
@@ -190,6 +190,34 @@ func (api *CloudAPI) UserCredentials(args params.UserClouds) (params.StringsResu
 	return results, nil
 }
 
+// AddCredential adds a new credential.
+// In contrast to UpdateCredentials() below, the new credential can be
+// for a cloud that the controller does not manage (this is required
+// for CAAS models)
+func (api *CloudAPI) AddCredential(args params.AddCredentialArgs) error {
```

Comment:
For better or worse, our policy is that all APIs should support bulk operations (on the server side at least). Note how `UpdateCredentials` supports multiple credentials and returns a struct with the error result for each. 

This makes the API implementations more complex/tedious but allows for more efficient use of the API.